articles/150_message_field_names.md

```diff
@@ -0,0 +1,94 @@
+---
+layout: default
+title: Message Field Name Conflicts
+permalink: articles/message_field_conflicts.html
+abstract:
+  As we move to support more platforms there are we are running into predefined fields and constants that conflict with languages or implementation keywords and macros.
+  This defines how to deal with existing and potential future conflicts.
+published: true
+author: '[Tully Foote](https://github.com/tfoote)'
+categories: Interfaces
+---
+
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Scope
+
+This article specifies how we will deal with conflicting fields and constants for different languages.
+In the use of the IDL (TODO (tfoote) reference) there will inevitably be conflicting symbols, keywords, or definitions.
+
+## Background and Motivation
+
+This has come up specifically as we're adding support for Windows for ROS2.
+`winnt.h` defines several macros that conflict with existing enumerations. 
+It is also expected to happen as support for new languages are added.
+We cannot know all the potential future keywords and restrictions from a language which is selected to add support for in the future so we must have a generic solution which will allow future languages to be added without disrupting the existing usages.
+
+## New constants
+
+To avoid conflicts generally all constants should be prefixed with a `k`. This will avoid almost all conflicts.
+
+Google Style Guide: https://google.github.io/styleguide/cppguide.html#Constant_Names
```

Comment:
I think this kind of section is too language specific. Instead I would phrase it that each language generator has to apply "some" mangling to avoid collisions with keywords as well as other symbols (like platform specific defines).

And in general the generated code will follow the style guide for that specific language. So I wouldn't tangle it with that topic here.

Reply:
Do you want the constants to be all upper case in the IDL file and then the c++ generator renames `FOO` to `kFOO`? I kind of feel like that's going to end up with a lot of divergance and decrease readability if the field names differ between implementations.  `foo.kCONSTANT` would end up as `foo.CONSTANT` in python.

Reply:
I didn't see this comment when I added my comment below in the second round of review.

The point that the `k` prefix is a C++ specific decision. What if another language comes by where it is recommended to use `j` as the prefix? Why do we favor one over the other? Therefore I think the IDL should be agnostic to that and any mangling should be handled in a language specific way even if that means that the constant might have different names in different languages.